tm2/pkg/store/immut/store.go

```diff
@@ -18,32 +18,32 @@ func New(parent types.Store) immutStore {
 }
 
 // Implements Store
-func (is immutStore) Get(key []byte) []byte {
+func (is immutStore) Get(key []byte) ([]byte, error) {
 	return is.parent.Get(key)
 }
 
 // Implements Store
-func (is immutStore) Has(key []byte) bool {
+func (is immutStore) Has(key []byte) (bool, error) {
 	return is.parent.Has(key)
 }
 
 // Implements Store
-func (is immutStore) Set(key, value []byte) {
+func (is immutStore) Set(key, value []byte) error {
 	panic("unexpected .Set() on immutStore")
```

Comment:
Do we need to leave these panics here?
The assumption is, because this is implementing an interface, the error would be handled top-level and not panic in case of the underlying type. What I'm trying to say is the user of this method does not need to know the underlying implementation in order to call this method (and know it will panic)

I suggest returning errors here, because it's an interface method implementation instead of panicking

Reply:
You are right, we should return errors here.